### PR TITLE
implement PNC artifacts download and fetch-artifacts-schema

### PR DESCRIPTION
**Pull request by hjmodi**
Story MMENG-1252
Epic MMENG-1282

# Maintainers will complete the following section

- [x] Commit messages are descriptive enough
- [x] Code coverage from testing does not decrease and new code is covered
- [x] JSON/YAML configuration changes are updated in the relevant schema
- [x] Changes to metadata also update the documentation for the metadata
- [x] Pull request has a link to an osbs-docs PR for user documentation updates
- [x] New feature can be disabled from a configuration file

### CODE REVIEW

File: atomic_reactor/utils/pnc.py [hunk 1]
```diff
@@ -18,9 +19,43 @@ def __init__(self, pnc_map, session=None):
...
-        # using urljoin here causes the API path in the base_api_url to be removed
-        self.get_scm_archive_request_url = self.base_api_url + '/' + pnc_map['get_scm_archive_path']
+
+        self.pnc_map = pnc_map
+        self.base_api_url = self.pnc_map['base_api_url']
+        self._get_artifact_url = None
+        self._get_scm_archive_request_url = None
+
+    @property
+    def get_artifact_url(self):
```
Comment (MartinBasti) [inline atomic_reactor/utils/pnc.py:29]: if this is a property, then is should be called just `artifact_url`

File: atomic_reactor/utils/pnc.py [hunk 2]
```diff
@@ -18,9 +19,43 @@ def __init__(self, pnc_map, session=None):
...
+        if not self._get_artifact_url:
+            # using urljoin here causes the API path in the base_api_url to be removed
+            self._get_artifact_url = self.base_api_url + '/' + self.pnc_map['get_artifact_path']
+        return self._get_artifact_url
+
+    @property
+    def get_scm_archive_request_url(self):
```
Comment (MartinBasti) [inline atomic_reactor/utils/pnc.py:36]: just `scm_archive_request_url`

File: atomic_reactor/utils/pnc.py (hunk 1 above)
Comment (hjmodi) [reply atomic_reactor/utils/pnc.py]: updated.

File: atomic_reactor/utils/pnc.py (hunk 2 above)
Comment (hjmodi) [reply atomic_reactor/utils/pnc.py]: updated.

**Comment by hjmodi** [comment]
Created https://github.com/containerbuildsystem/osbs-docs/pull/183

**Review by mkosiarc — APPROVED** [review]

**Review by MartinBasti — APPROVED** [review]

**Comment by hjmodi** [comment]
release notes updated

**Pull request merged**